Changes PSA key storage format to include key bits (#3748)
ARMmbed/mbedtls: ChangeLog.d/systematically_store_bit_size_3740.txt

Changes
   * Always store key bits attribute in PSA key storage. Previously stored
     keys should be augmented with the bits value to be compatible with the
     updated version. Also removes the stored bits field from the storage
     format for the dynamic Secure Element driver. Previously stored keys

>      must be updated to remove the stored bits field. #3740

Reviewer: This is both too detailed and not detailed enough. The changelog file is intended for library users. The short version is that we don't provide an upgrade path. To upgrade an existing device, you need to either wipe out the storage, or work out how to upgrade existing key objects from the format description, you can't work out what to do just from the changelog entry.

Author: I have attempted a re-write now that gives the same general guideline as in your comment, and a pointer to the format documentation.

Changes
   * Always store key bits attribute in PSA key storage. Previously stored
     keys should be augmented with the bits value to be compatible with the
     updated version. #3740

>    * Removes the stored bits field from the storage format for the

Reviewer: This is really a single change: store the key size systematically instead of just for dynamic secure element keys.

>    * PSA persistent storage format is updated to always store the key bits
>      attribute. No automatic upgrade path is provided. Previously stored keys
>      must be erased, or manually upgraded based on the key storage format
>      specification (mbed-crypto-storage-specification.md). The storage format
>      for the dynamic Secure Element driver has also been updated. Previously
>      stored keys must be erased, or manually upgraded. #3740

Reviewer: The part about secure elements is redundant.
```suggestion
   * The PSA persistent storage format is updated to always store the key bits
     attribute. No automatic upgrade path is provided. Previously stored keys
     must be erased, or manually upgraded based on the key storage format
     specification (docs/architecture/mbed-crypto-storage-specification.md). #3740
```

It's very likely that we'll also upgrade the format version to 1 before the next release anyway.

Author: Ok, thanks for the suggestion.
